src/formats/png.zig

```diff
@@ -63,9 +74,137 @@ pub const PNG = struct {
     }
 
     pub fn writeImage(allocator: Allocator, write_stream: *Image.Stream, image: Image, encoder_options: Image.EncoderOptions) ImageWriteError!void {
-        _ = allocator;
-        _ = write_stream;
-        _ = image;
-        _ = encoder_options;
+        const options = encoder_options.png;
+        
+        var writer = write_stream.writer();
+
+        try writeSignature(writer);
+        try writeHeader(writer, image, options);
+        if (image.pixelFormat().isIndex()) {
+            try writePalette(writer, image);
+            try writeTransparencyInfo(writer, image); // TODO: pixel format where there is no transparency
+        }
+        try writeData(allocator, writer, image, options);
+        try writeTrailer(writer);
+    }
+
+    fn writeSignature(writer: anytype) !void {
+        try writer.writeAll("\x89\x50\x4E\x47\x0D\x0A\x1A\x0A");
+    }
+
+    // IHDR
+    fn writeHeader(writer: anytype, image: Image, encoder_options: EncoderOptions) ImageWriteError!void {
+        // TODO: interlaced png files
+        if (encoder_options.interlaced)
+            return ImageWriteError.Unsupported;
+
+        // Color type byte
+        const color_type: u8 = switch (image.pixels) {
+            .rgb24,
+            .rgb48 => 2,
+
+            .rgba32,
+            .rgba64 => 6,
+
+            .grayscale1,
+            .grayscale2,
+            .grayscale4,
+            .grayscale8,
+            .grayscale16 => 0,
+
+            .grayscale8Alpha,
+            .grayscale16Alpha =>  4,
+
+            .indexed1,
+            .indexed2,
+            .indexed4,
+            .indexed8 => 3,
+            // TODO: indexed without alpha
+            else => return ImageWriteError.Unsupported,
+        };
+
+        // Bits per sample layer
+        const bps: u8 = image.pixelFormat().bitsPerChannel();
```

Comment:
`bps` is not readable enough, please use `bits_per_channel`. Acronyms are never good for code readability.